rust/agents/relayer/src/msg/serial_submitter.rs

```diff
@@ -54,329 +55,243 @@ use super::{
+    prepare_queue: OpQueue,
+    confirm_queue: OpQueue,
+    metrics: SerialSubmitterMetrics,
+) -> Result<()> {
+    while let Some(mut op) = rx_submit.recv().await {
+        trace!(?op, "Submitting operation");
+        match op.submit().await {
+            PendingOperationResult::Success => {
+                debug!(?op, "Operation submitted");
+                metrics.ops_submitted.inc();
+                confirm_queue.lock().await.push(Reverse(op));
+            }
+            PendingOperationResult::NotReady => {
+                panic!("Pending operation was prepared and therefore must be ready")
+            }
+            PendingOperationResult::Reprepare => {
+                metrics.ops_failed.inc();
+                prepare_queue.lock().await.push(Reverse(op));
+            }
+            PendingOperationResult::Drop => {
+                metrics.ops_submitted.inc();
+            }
+            PendingOperationResult::CriticalFailure(e) => return Err(e),
         }
+    }
+    bail!("Internal submitter channel was closed");
+}
 
-        let ism_address = self
-            .mailbox
-            .recipient_ism(msg.message.recipient)
-            .await
-            .context(CTX)?;
-
-        let Some(metadata) = self.metadata_builder
-            .build(ism_address, &msg.message)
-            .await?
-        else {
-            info!("Could not fetch metadata");
-            return Ok(false)
-        };
-
-        // Estimate transaction costs for the process call. If there are issues, it's
-        // likely that gas estimation has failed because the message is
-        // reverting. This is defined behavior, so we just log the error and
-        // move onto the next tick.
-        let tx_cost_estimate = match self
-            .mailbox
-            .process_estimate_costs(&msg.message, &metadata)
-            .await
-        {
-            Ok(tx_cost_estimate) => tx_cost_estimate,
-            Err(error) => {
-                info!(?error, "Error estimating process costs");
-                return Ok(false);
-            }
+#[instrument(skip_all)]
+async fn confirm_task(
+    prepare_queue: OpQueue,
+    confirm_queue: OpQueue,
+    metrics: SerialSubmitterMetrics,
+) -> Result<()> {
+    loop {
+        // Pick the next message to try confirming.
+        let next = {
+            let mut queue = confirm_queue.lock().await;
+            metrics.confirm_queue_length.set(queue.len() as i64);
+            queue.pop()
         };
-
-        // If the gas payment requirement hasn't been met, move to the next tick.
-        let Some(gas_limit) = self
-            .gas_payment_enforcer
-            .message_meets_gas_payment_requirement(&msg.message, &tx_cost_estimate)
-            .await?
-        else {
-            info!(?tx_cost_estimate, "Gas payment requirement not met yet");
-            return Ok(false);
+        let Some(Reverse(mut op)) = next else {
+            sleep(Duration::from_secs(60)).await;
```

Comment:
nit: Consider adding a comment as to why sleeping so long here makes sense

Reply:
Better to decrease actually now that I think about it